For GM965, reserve a buffer of unused memory between the exa offscreen buffer and end of aperture
Prevents freeze with exa on the GM965, LP: #359392
Index: xserver-xorg-video-intel-2.6.3/src/i830_memory.c
===================================================================
--- xserver-xorg-video-intel-2.6.3.orig/src/i830_memory.c	2009-05-19 22:07:40.000000000 +0200
+++ xserver-xorg-video-intel-2.6.3/src/i830_memory.c	2009-05-19 22:13:36.000000000 +0200
@@ -501,6 +501,10 @@
 	 */
 	mmsize = size;
 
+	/* GM965 needs some unused memory above the exa offscreen buffer */
+	if ((pI830->accel == ACCEL_EXA) && IS_I965G(pI830)) {
+	    mmsize -= ROUND_TO_PAGE(512 * pI830->cpp * pScrn->virtualY);
+	}
 	/* EXA area is fixed. */
 	if (pI830->accel == ACCEL_EXA) {
 	    mmsize -= ROUND_TO_PAGE(3 * pScrn->displayWidth * pI830->cpp *
